File: case-developer-java/src/main/java/com/befrank/casedeveloperjava/models/Participant.java
package com.befrank.casedeveloperjava.models;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "CASE_DEELNEMERS")
public class Participant {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    private String name;
    private String address;
    private String placeOfBirth;
    private boolean isEmployed;
    private LocalDate dateOfBirth;
    private String email;
    private Employment employment;
    private long pensionAccountNumber;

}

File: case-developer-frontend/src/app/models/participant.ts
import {Employment} from "./employment";

export class Participant {
    id: number;
    name: string;
    address: string;
    placeOfBirth: string;
    isEmployed: boolean;
    dateOfBirth: Date | null;
    email: string;
    employment: Employment;
    pensionAccountNumber: number;
    pension: number;


    constructor() {
        this.id = 0;
        this.name = '';
        this.address = '';
        this.placeOfBirth = '';
        this.isEmployed = false;
        this.dateOfBirth = null;
        this.email = '';
        this.employment = new Employment();
        this.pensionAccountNumber = 0;
        this.pension = 0;
    }
}
